### PR TITLE
Resource Factory & listener

### LINKED ISSUE #646: Factory listener

### LINKED ISSUE #647: Add Factory arguments

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | Fixes #646 & Fixes #647
| License         | MIT

### CODE REVIEW

File: src/Component/Factory/ArgumentParser.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+namespace Sylius\Component\Resource\Factory;
+
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Security\Core\Authentication\Token\NullToken;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+
+final class ArgumentParser
```
Comment (lchrusciel) [inline src/Component/Factory/ArgumentParser.php]: Missing interface. I would prefer to not merge it without it

Comment (loic425) [reply src/Component/Factory/ArgumentParser.php]: done :)

File: src/Bundle/Resources/config/services/listener.xml [hunk 2]
```diff
@@ -28,6 +28,13 @@
             <tag name="kernel.event_listener" event="kernel.request" priority="4" />
         </service>
 
+        <service id="sylius.listener.factory" class="Sylius\Component\Resource\Symfony\EventListener\FactoryListener">
+            <argument type="service" id="sylius.resource_metadata_operation.initiator.http_operation" />
+            <argument type="service" id="sylius.context.initiator.request_context" />
+            <argument type="service" id="sylius.state.factory" />
+            <tag name="kernel.event_listener" event="kernel.request" priority="3" />
```
Comment (lchrusciel) [inline src/Bundle/Resources/config/services/listener.xml:35]: We can reconsider these priorities in future implementation

File: src/Component/State/FactoryInterface.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Resource\State;
+
+use Sylius\Component\Resource\Context\Context;
+use Sylius\Component\Resource\Metadata\Operation;
+
+/**
+ * @experimental
```
Comment (lchrusciel) [inline src/Component/State/FactoryInterface.php:20]: Why do we have tag only here?

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
Thank you, @loic425!

File: src/Bundle/Resources/config/services/listener.xml (hunk 2 above)
Comment (loic425) [reply src/Bundle/Resources/config/services/listener.xml:35]: Hell yes.

File: src/Component/State/FactoryInterface.php (hunk 3 above)
Comment (loic425) [reply src/Component/State/FactoryInterface.php:20]: Yes I think that'd be better to add experimental at least on every interfaces.

**Pull request merged**